apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexOfferProcessor.java

```diff
@@ -0,0 +1,223 @@
+                .recipient(accountId)
+                .status(STEP_1.ordinal())
+                .build());
+
+        for (ExchangeContract incomeContract : incomeContracts) {
+            try {
+                DexOffer offer = dexService.getOfferByTransactionId(incomeContract.getOrderId());
+                DexOffer counterOffer = dexService.getOfferByTransactionId(incomeContract.getCounterOrderId());
+
+                if(!counterOffer.getStatus().isOpen() || !isContractStep1Valid(incomeContract)){
+                    continue;
+                }
+
+                String passphrase = secureStorageService.getUserPassPhrase(accountId);
+                CreateTransactionRequest transferMoneyReq = CreateTransactionRequest
+                        .builder()
+                        .passphrase(passphrase)
+                        .deadlineValue("1440")
+                        .publicKey(Account.getPublicKey(accountId))
+                        .senderAccount(Account.getAccount(accountId))
+                        .keySeed(Crypto.getKeySeed(Helper2FA.findAplSecretBytes(accountId, passphrase)))
+                        .broadcast(true)
+                        .recipientId(0L)
+                        .ecBlockHeight(0)
+                        .ecBlockId(0L)
+                        .build();
+
+                String txId = dexService.transferMoneyWithApproval(transferMoneyReq, counterOffer, offer.getToAddress(), incomeContract.getSecretHash(), STEP_2);
+
+                if (txId == null) {
+                    throw new AplException.ExecutiveProcessException("Transfer money wasn't finish success. Orderid: " + incomeContract.getOrderId() + ", counterOrder:  " + incomeContract.getCounterOrderId() + ", " + incomeContract.getContractStatus());
+                }
+
+                DexContractAttachment contractAttachment = new DexContractAttachment(counterOffer.getTransactionId(), offer.getTransactionId(), null, txId, null, STEP_2);
+                //TODO move it to some util
+                CreateTransactionRequest createTransactionRequest = CreateTransactionRequest
+                        .builder()
+                        .passphrase(passphrase)
+                        .publicKey(Account.getPublicKey(accountId))
+                        .senderAccount(Account.getAccount(accountId))
+                        .keySeed(Crypto.getKeySeed(Helper2FA.findAplSecretBytes(accountId, passphrase)))
+                        .deadlineValue("1440")
+                        .feeATM(Constants.ONE_APL * 2)
+                        .broadcast(true)
+                        .recipientId(0L)
+                        .ecBlockHeight(0)
+                        .ecBlockId(0L)
+                        .attachment(contractAttachment)
+                        .build();
+
+                Transaction transaction = dexOfferTransactionCreator.createTransaction(createTransactionRequest);
+
+                if (transaction == null) {
```

Comment:
Is it possible to get here null?

Reply:
yes, if we will catch an exception during validating/creating/sending a transaction.